ibis/tests/expr/test_window_functions.py

```diff
@@ -38,3 +38,27 @@ def test_mutate_with_analytic_functions(alltypes):
     for field in proj.op().selections[1:]:
         assert isinstance(field, ops.Alias)
         assert isinstance(field.arg, ops.WindowFunction)
+
+
+def test_value_over_api(alltypes):
+    t = alltypes
+
+    w1 = ibis.window(rows=(0, 1), group_by=t.g, order_by=[t.f, t.h])
+    w2 = ibis.window(range=(-1, 1), group_by=[t.g, t.a], order_by=[t.f])
+    w3 = ibis.window(
+        between=(ibis.interval(days=1), ibis.interval(days=5)), order_by=[t.f]
+    )
+
+    expr1 = t.f.cumsum().over(rows=(0, 1), group_by=t.g, order_by=[t.f, t.h])
+    expected1 = t.f.cumsum().over(w1)
+    assert expr1.equals(expected1)
+
+    expr2 = t.f.cumsum().over(range=(-1, 1), group_by=[t.g, t.a], order_by=[t.f])
+    expected2 = t.f.cumsum().over(w2)
+    assert expr2.equals(expected2)
+
+    expr3 = t.f.cumsum().over(
+        between=(ibis.interval(days=1), ibis.interval(days=5)), order_by=[t.f]
```

Comment:
I'm not sure that we should expose the between keyword argument, it enables inferring the window frame kind if it isn't passed explicitly.

I tend to think that we should enforce either `over(rows=` or `over(range=)` but not `over(between=)`.